initial commit to address GH-ISSUE #1

codylane opened the pull request:
As promised here is an implementation that uses pure STDLIB ruby instead of shelling out to `whoami`.

Also, I took the liberty to add 100% test coverage to the method in question.

* [x] - Stubbed out an initial commit of integration tests using Vagrant.  I'll add serverspec  tests to it in the next few days.

codylane commented:
Ok, I believe I've addressed the open issues for this PR now. Please let me know if you have questions or concerns.

dustymabe commented:
Wow @codylane - that's a lot of awesome work. Sorry I'm only looking at it now.

For this particular issue can you remove all the extra stuff to another PR and just focus on fixing #1?

codylane commented:
> Wow @codylane - that's a lot of awesome work. Sorry I'm only looking at it now.
> 
> For this particular issue can you remove all the extra stuff to another PR and just focus on fixing #1?

You are very welcome, I'm excited and committed to help the open source community as best as I can.  If this PR is approved I promise to be a maintainer for anything I help create.

No problem, I'd be happy to just address #1. Not sure I can remove the commits so I may need to close this one and re-open a new PR.

dustymabe commented:
> No problem, I'd be happy to just address #1. Not sure I can remove the commits so I may need to close this one and re-open a new PR.

You can easily remove the commits with an [interactive rebase](https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d). It's the best thing since sliced bread :)

codylane commented:
I won't lie, I avoid doing interactive rebase as not only are they are to follow but they are hard to do right without additional problems and or messing up your git history or reflog.  

Instead, I use the reflog to tell me which commits I want and then I re-write the history in a separate temporary working directory.  It doesn't make what I do right, it just easier for me to understand and I've had much better luck than trying to pick commits or squash the history.   Just my 3$.

I've removed the CI/CD stuff as you recommended and updated this commit to only reflect stuff needed to get GH #1 to pass. I'm still not sure this will be to your liking but it's my final attempt.

codylane commented:
Closing the request as it has been almost 60+ days without traction or guidance of how to proceed.  I've lot all motivation for trying to help.  I wish you the best. Take care.
